File: mwm_publishing/src/main/java/com/mwm/android/publishing/StoreActivity.java
package com.mwm.android.publishing;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.content.res.Resources;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.widget.ImageView;

public class StoreActivity extends AppCompatActivity {

    public static void start(Context context) {
        Intent intent = new Intent(context, StoreActivity.class);
        if (!(context instanceof Activity)) {
            intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TOP);
        }
        context.startActivity(intent);
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_store);
        ImageView image = findViewById(R.id.activity_store_image_background);
        setImageByDrawableName(image, "mwm_store_background_by_name");
    }

    private static void setImageByDrawableName(ImageView imageView, String resourceName) {
        Context context = imageView.getContext();
        Resources resources = context.getResources();
        String packageName = context.getPackageName();
        int id = resources.getIdentifier(resourceName, "drawable", packageName);
        imageView.setImageResource(id);
    }
}

File: app/src/main/java/com/mercandalli/android/browser/store/StoreActivity.kt
package com.mercandalli.android.browser.store

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.mercandalli.android.browser.R
import com.mercandalli.android.browser.monetization.Monetization
import com.mercandalli.android.browser.monetization.MonetizationGraph
import com.mercandalli.android.browser.in_app.InAppManager

class StoreActivity : AppCompatActivity(),
        StoreContract.Screen {

    private val userAction = createUserAction()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_store)

        findViewById<View>(R.id.activity_store_buy_subscription).setOnClickListener {
            userAction.onBuySubscriptionClicked(object : InAppManager.ActivityContainer {
                override fun get() = this@StoreActivity
            })
        }
        val monetization = Monetization.fromJson(intent.extras.getString(EXTRA_MONETIZATION))
        userAction.onCreate(monetization)
    }

    private fun createUserAction(): StoreContract.UserAction {
        val inAppManager = MonetizationGraph.getInAppManager()
        return StorePresenter(
                this,
                inAppManager
        )
    }

    companion object {
        private const val EXTRA_MONETIZATION = "StoreActivity.Extra.EXTRA_MONETIZATION"

        @JvmStatic
        fun start(context: Context, monetization: Monetization) {
            val intent = Intent(context, StoreActivity::class.java)
            if (context !is Activity) {
                intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TOP
            }
            intent.putExtra(EXTRA_MONETIZATION, Monetization.toJson(monetization))
            context.startActivity(intent)
        }
    }
}